### PR TITLE
Renaming Wepback config objects and using const/let

### PR DESCRIPTION
**Pull request by Golodhros**
## Description

Moves the webpack config file into ES6, renaming some objects to be more consistent with Britecharts-React config.

## Motivation and Context

First step on https://github.com/eventbrite/britecharts/issues/461

## How Has This Been Tested?

Ran tests and documentation

## Types of changes

- [x] Refactor (changes the way we code something without changing its functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by dalerasrorov-eb** [comment]
LGTM

File: webpack.config.js [hunk 1]
```diff
@@ -1,86 +1,88 @@
...
+    enforce: 'pre',
+    loader: 'eslint-loader',
+    options: {
+        emitWarning: true,
+        failOnError: false,
+    }
+};
 
-    plugins = [
+let plugins = [
```
Comment (amber-eb) [inline webpack.config.js]: Why `let` for this one?

File: webpack.config.js [hunk 2]
```diff
@@ -1,86 +1,88 @@
...
+// Move to Parts
+const babelLoader = {
+    test: /\.js$/,
+    exclude: /(node_modules)/,
+    use: [{
+        loader: 'babel-loader',
         options: {
-            emitWarning: true,
-            failOnError: false,
+            presets: ['es2015', 'stage-0'],
```
Comment (amber-eb) [inline webpack.config.js]: What does the`stage-0`preset do?

File: webpack.config.js [hunk 3]
```diff
@@ -1,86 +1,88 @@
...
+const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
+const merge = require('webpack-merge');
+
+const env = require('yargs').argv.mode;
+const isProduction = env === 'prod' || env === 'prodUMD';
+
+const projectName = 'britecharts';
+
+const CHARTS = {
+    'bar': './src/charts/bar.js',
```
Comment (amber-eb) [inline webpack.config.js]: Is it necessary for the keys to be strings? Would it be possible to do:
```
const CHARTS = {
    bar: './src/charts/bar.js',
```

Comment (miglesiasEB) [reply webpack.config.js]: Yeah, totally

File: webpack.config.js (hunk 2 above)
Comment (miglesiasEB) [reply webpack.config.js]: it's a Babel configuration: https://babeljs.io/docs/plugins/preset-stage-0/

File: webpack.config.js (hunk 1 above)
Comment (miglesiasEB) [reply webpack.config.js]: In line 91, if the environment is production, we push something in.

That said, a const won't break if we do that, but it feels weird.

File: webpack.config.js (hunk 3 above)
Comment (amber-eb) [reply webpack.config.js]: Just to clarify, what I am asking is whether it is necessary for the keys to be in quotes.

File: webpack.config.js (hunk 1 above)
Comment (amber-eb) [reply webpack.config.js]: I see, so this conveys intent to change the values inside of the array.

Comment (dalerasrorov-eb) [reply webpack.config.js]: I think it is a good convention to declare variables with `let` if they are being mutated in the future (unless they are 3rd party modules or a custom module of some sort.)

**Pull request merged**